recipe/install_nvcc.sh

```diff
@@ -90,19 +90,24 @@ export CMAKE_ARGS="\${CMAKE_ARGS}"
 
 ### /CMake configurations
 
-mkdir -p "\${CONDA_BUILD_SYSROOT}/lib"
-
 # Add \$(libcuda.so) shared object stub to the compiler sysroot.
 # Needed for things that want to link to \$(libcuda.so).
 # Stub is used to avoid getting driver code linked into binaries.
 
-# Make a backup of \$(libcuda.so) if it exists
-if [[ -f "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" ]]
+if [[ ! -z "\${CONDA_BUILD_SYSROOT+x}" ]]
 then
+  mkdir -p "\${CONDA_BUILD_SYSROOT}/lib"
+  # Make a backup of \$(libcuda.so)
   LIBCUDA_SO_CONDA_NVCC_BACKUP="\${CONDA_BUILD_SYSROOT}/lib/libcuda.so-conda-nvcc-backup"
-  mv "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" "\${LIBCUDA_SO_CONDA_NVCC_BACKUP}"
+  if [[ -f "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" ]]
+  then
+    mv -f "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" "\${LIBCUDA_SO_CONDA_NVCC_BACKUP}"
+  fi
+  ln -s "\${CUDA_HOME}/lib64/stubs/libcuda.so" "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so"
```

Comment:
I believe there's some OS package managers where the toolkit gets installed into the standard system library / include locations and I'm not sure this is the location of where the stubs end up. It would be nice to verify that this stub library exists and warn or error if it doesn't.

Reply:
That makes sense.

Though it is worth noting we do [raise]( https://github.com/conda-forge/nvcc-feedstock/blob/570b94fae039161e3c043d2b255e293da351e31a/recipe/install_nvcc.sh#L60-L64 ) if `libcuda.so` is not found.

Also wouldn't it be possible for the user to override `$CUDA_HOME`? If the user doesn't do that, wouldn't [our `cuda-gdb` detection logic]( https://github.com/conda-forge/nvcc-feedstock/blob/570b94fae039161e3c043d2b255e293da351e31a/recipe/install_nvcc.sh#L44 ) find the right location (IOW does `cuda-gdb` also land in a system location when `libcuda.so` does)?

Reply:
Ah good point. Yeah happy to follow up on these deployment issues on a case-by-case basis as they come up